geoh5py/objects/text.py

```diff
@@ -0,0 +1,192 @@
+#  it under the terms of the GNU Lesser General Public License as published by '
+#  the Free Software Foundation, either version 3 of the License, or           '
+#  (at your option) any later version.                                         '
+#                                                                              '
+#  geoh5py is distributed in the hope that it will be useful,                  '
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of              '
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               '
+#  GNU Lesser General Public License for more details.                         '
+#                                                                              '
+#  You should have received a copy of the GNU Lesser General Public License    '
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.           '
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+
+from __future__ import annotations
+
+import uuid
+from json import loads
+from typing import TYPE_CHECKING
+
+import numpy as np
+from pydantic import (
+    AliasChoices,
+    AliasGenerator,
+    BaseModel,
+    ConfigDict,
+    Field,
+    field_validator,
+)
+from pydantic.alias_generators import to_pascal, to_snake
+
+from .points import Points
+
+
+if TYPE_CHECKING:
+    from numpy import ndarray
+
+
+class TextEntry(BaseModel):
+    """
+    Core parameters for text mesh data.
+    """
+
+    model_config = ConfigDict(
+        alias_generator=AliasGenerator(  # type: ignore
+            validation_alias=to_snake,
+            serialization_alias=to_pascal,
+        )
+    )
+    starting_point: str
+    annotation: int = 0
+    color: str = "#ffffff"
+    direction: str = "{1,0,0}"
+    font: str = "builtin_simple"
+    font_size: int = 10
+    layer_name: str = ""
+    normal: str = "{0,0,1}"
+    obliquing_angle: int = 0
+    rotation_angle: int = 0
+    text: str = "<no-data>"
+    width_scale_factor: float = 1.0
+
+    @field_validator("starting_point", "normal", "direction", mode="before")
+    @classmethod
+    def array_to_str(cls, float_list: np.ndarray | list):
+        if isinstance(float_list, np.ndarray | list):
+            float_list = "{" + ",".join([str(val) for val in float_list]) + "}"
+
+        return float_list
+
+
+class TextData(BaseModel):
+    """
+    Text mesh data.
+    """
+
+    text_data: list[TextEntry] = Field(
+        validation_alias=AliasChoices("text_data", "Text Data"),
+        serialization_alias="Text Data",
+    )
+
+
+class TextObject(Points):
+    """
+    Text object for annotated text labels in viewport.
+    """
+
+    _attribute_map: dict = Points._attribute_map.copy()
+    _attribute_map.update({"TextMesh Data": "text_mesh_data"})
+    _default_name = "Text Object"
+    _TYPE_UID: uuid.UUID = uuid.UUID("{c00905d1-bc3b-4d12-9f93-07fcf1450270}")
```

Comment:
_Type_uid has different implementation for different class:

eg:

in geo_image: 
```
    _TYPE_UID = uuid.UUID(
        fields=(0x77AC043C, 0xFE8D, 0x4D14, 0x81, 0x67, 0x75E300FB835A)
    )
```

would be nice to make it the same in every classes  